packages/plugins/typescript/resolvers/src/index.ts

```diff
@@ -72,6 +72,13 @@ export type NewStitchingResolver<TResult, TParent, TContext, TArgs> = {
       context: TContext,
       info: GraphQLResolveInfo
     ) => Promise<TResult> | TResult;`);
+
+    defsToInclude.push(`
+      type scalarCheck<T, S> = S extends true ? T : nullableCheck<T, S>;
+      type nullableCheck<T, S> = Maybe<T> extends T ? Maybe<listCheck<NonNullable<T>, S>> : listCheck<T, S>;
+      type listCheck<T, S> = T extends (infer U)[] ? nullableCheck<U, S>[] : GraphQLRecursivePick<T, S>;
+      export type GraphQLRecursivePick<T, S> = { [K in keyof T & keyof S]: scalarCheck<T[K], S[K]> };
+    `);
```

Comment:
<3

Reply:
Maybe just change it to be pascal case? (for example, `ScalarCheck` instead `scalarCheck`) to it will aligned with other helpers we have in this repo.